src/usecases/configure-figma-teams-use-case.test.ts

```diff
@@ -0,0 +1,77 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { type FigmaTeam, FigmaTeamAuthStatus } from '../domain/entities';
+import { generateConnectInstallation } from '../domain/entities/testing';
+import { figmaService } from '../infrastructure/figma';
+import { figmaTeamRepository } from '../infrastructure/repositories';
+
+import { configureFigmaTeamUseCase } from '.';
+
+describe('configureFigmaTeamUseCase', () => {
+	it('should create a webhook and FigmaTeam record', async () => {
+		const teamId = uuidv4();
+		const atlassianUserId = uuidv4();
+		const connectInstallation = generateConnectInstallation();
+		const webhookId = uuidv4();
+
+		jest
+			.spyOn(figmaService, 'createFileUpdateWebhook')
+			.mockResolvedValue({ teamId, webhookId });
+		jest
+			.spyOn(figmaTeamRepository, 'upsert')
+			.mockResolvedValue({} as FigmaTeam);
+
+		await configureFigmaTeamUseCase.execute(
+			teamId,
+			atlassianUserId,
+			connectInstallation,
+		);
+
+		expect(figmaService.createFileUpdateWebhook).toBeCalledWith(
+			teamId,
+			atlassianUserId,
+			connectInstallation.sharedSecret,
+		);
+		expect(figmaTeamRepository.upsert).toBeCalledWith({
+			webhookId,
+			teamId,
+			teamName: 'TODO',
+			figmaAdminAtlassianUserId: atlassianUserId,
+			authStatus: FigmaTeamAuthStatus.OK,
+			connectInstallationId: connectInstallation.id,
+		});
+	});
+
+	it('should throw if creating the webhook fails', async () => {
+		const error = new Error('create webhook failed');
+		jest
+			.spyOn(figmaService, 'createFileUpdateWebhook')
+			.mockRejectedValue(error);
+
+		await expect(
+			configureFigmaTeamUseCase.execute(
+				uuidv4(),
+				uuidv4(),
+				generateConnectInstallation(),
+			),
+		).rejects.toStrictEqual(error);
+	});
+
+	it('should throw if creating the FigmaTeam record fails', async () => {
```

Comment:
**suggestion:** Verifying that out code does not catch an exception can be excessive since this is the default behavior, which does not require any special considerations. Consider testing only those error scenario where we _handle_ an error (e.g., try to recover from a failure, translate an error, suppress an error, etc.). Otherwise, we would need to add a test for almost any call of a built-in or our function (since many of them can throw an error).

For `configureFigmaTeamUseCase`, we might also need to test that a `FigmaTeam` record is _not_ created if a webhook creation fails (which is important from the business perspective).